server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -310,17 +312,25 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
             return;
         }
 
-        cancellable.checkCancelled();
-        weight = wrapWeight(weight);
-        // See please https://github.com/apache/lucene/pull/964
-        collector.setWeight(weight);
         final LeafCollector leafCollector;
         try {
+            cancellable.checkCancelled();
+            weight = wrapWeight(weight);
+            // See please https://github.com/apache/lucene/pull/964
+            collector.setWeight(weight);
             leafCollector = collector.getLeafCollector(ctx);
         } catch (CollectionTerminatedException e) {
             // there is no doc of interest in this reader context
             // continue with the following leaf
             return;
+        } catch (QueryPhase.TimeExceededException e) {
+            // Swallow timeout exception and return
+            if (searchContext.request().allowPartialSearchResults() == false) {
+                // Can't rethrow TimeExceededException because not serializable
+                throw new QueryPhaseExecutionException(searchContext.shardTarget(), "Time exceeded");
+            }
+            searchContext.queryResult().searchTimedOut(true);
```

Comment:
AFAIK, searchContext (and searchContext.queryResult()) was not supposed to be modified from multiple threads but that would certainly be the case here.

Reply:
This is a really good point. Let me double check the consumers of `searchTimedOut` -- since we are only modifying the value to false it may not matter if the change is immediately visible to other threads.